crates/factor-outbound-http/src/wasi.rs

```diff
@@ -44,16 +48,140 @@ use crate::{
     wasi_2023_10_18, wasi_2023_11_10, InstanceState, OutboundHttpFactor, SelfRequestOrigin,
 };
 
+const DEFAULT_TIMEOUT: Duration = Duration::from_secs(600);
+
 pub(crate) struct HasHttp;
 
 impl HasData for HasHttp {
     type Data<'a> = WasiHttpImpl<WasiHttpImplInner<'a>>;
 }
 
+impl p3::WasiHttpCtx for InstanceState {
+    fn send_request(
+        &mut self,
+        request: http::Request<BoxBody<Bytes, p3_types::ErrorCode>>,
+        options: Option<p3::RequestOptions>,
+        fut: Box<dyn Future<Output = Result<(), p3_types::ErrorCode>> + Send>,
+    ) -> Box<
+        dyn Future<
+                Output = Result<
+                    (
+                        http::Response<BoxBody<Bytes, p3_types::ErrorCode>>,
+                        Box<dyn Future<Output = Result<(), p3_types::ErrorCode>> + Send>,
+                    ),
+                    TrappableError<p3_types::ErrorCode>,
+                >,
+            > + Send,
+    > {
+        // TODO: do we neeed to do anything with `fut`?
```

Comment:
Do we know the status of this TODO?

Reply:
Never mind, I'm an idiot: this is outbound HTTP.  Sorry for the noise.

Reply:
> would that include guest errors produced after the response has been finished?

The guest can send an error to `fut` at any time, i.e. before, during, or after the response has arrived in full and been delivered to the guest; there's nothing in the API that constraints when or why the guest would do that.  In any case, there's not much we can do about it here 🤷 .